src/journal-remote/journal-gatewayd.c

```diff
@@ -641,6 +641,66 @@ static int request_handler_fields(
         return MHD_queue_response(connection, MHD_HTTP_OK, response);
 }
 
+static int output_field_name(FILE *f, OutputMode m, const char *d) {
+        if (m == OUTPUT_JSON)
+                return fprintf(f, "{ \"field\" : \"%s\" }\n", d);
```

Comment:
Writing json format by hand is not good. Please use our json library. src/shared/json.[ch].

Reply:
I do here the same assumption as in [L527](https://github.com/systemd/systemd/blob/63947fede8f2cf3cf22a4fe96832ea6f20c5df19/src/journal-remote/journal-gatewayd.c#L527) (`output_field`), that field names always match /[0-9A-Z_]*/ regex.

But actually I do not see any specification backing this assumption. It's probably safer to fix this in the two points.